.github/workflows/src/label-rules.js

```diff
@@ -0,0 +1,864 @@
+  // \private\azure-rest-api-specs-pipeline\config\production\Azure\azure-rest-api-specs-pr.yml
+  {
+    precedence: 2,
+    anyPrerequisiteLabels: ["CI-BreakingChange-Go"],
+    anyRequiredLabels: ["Approved-SdkBreakingChange-Go"],
+    troubleshootingGuide:
+      `Your PR has breaking changes in the generated SDK for go (label: <code>CI-BreakingChange-Go</code>). ` +
+      `${diagramTsg(3, true)}.`,
+  },
+  {
+    precedence: 2,
+    anyPrerequisiteLabels: ["CI-BreakingChange-Python-Track2"],
+    anyRequiredLabels: ["Approved-SdkBreakingChange-Python"],
+    troubleshootingGuide:
+      `Your PR has breaking changes in the generated SDK for python (label: <code>CI-BreakingChange-Python-Track2</code>). ` +
+      `${diagramTsg(3, true)}.`,
+  },
+  {
+    precedence: 2,
+    anyPrerequisiteLabels: ["CI-BreakingChange-JavaScript"],
+    anyRequiredLabels: ["Approved-SdkBreakingChange-JavaScript"],
+    troubleshootingGuide:
+      `Your PR has breaking changes in the generated SDK for javascript (label: <code>CI-BreakingChange-JavaScript</code>). ` +
+      `${diagramTsg(3, true)}.`,
+  },
+];
+
+/** @type {RequiredLabelRule[]} */
+const rulesPri3Blockers = [
+  {
+    precedence: 3,
+    anyPrerequisiteLabels: ["Approved-OkToMerge"],
+    anyRequiredLabels: [],
+    troubleshootingGuide:
+      "This PR has been approved with <code>Approved-OkToMerge</code> label.<br/>" +
+      "This label should be used only to approve PRs targeting private specs repo <code>main</code> branch.<br/>" +
+      "PRs targeting this branch cannot be merged, as this branch is a mirror of the public specs repo <code>main</code> branch.<br/>" +
+      "If you want to publish changes in this PR to the public repo, see: " +
+      `${href("aka.ms/azsdk/move-pr", "https://aka.ms/azsdk/move-pr")}.<br/>` +
+      "If this label was applied erroneously to a different branch than private specs repo <code>main</code> branch, remove the label.",
+  },
+];
+
+export const requiredLabelsRules = rulesPri0dataPlane
+  .concat(rulesPri0NotReadyForArmReview)
+  .concat(rulesPri0ArmRpaas)
+  .concat(rulesPri0Changes)
+  .concat(rulesPri0ArmRev)
+  .concat(rulesPri1ArmRev)
+  .concat(rulesPri1Suppressions)
+  .concat(rulesPri1ArcReview)
+  .concat(rulesPri2Sdk)
+  .concat(rulesPri2LegacySdk)
+  .concat(rulesPri3Blockers);
+
+/**
+ * @param {import('github-script').AsyncFunctionArguments} AsyncFunctionArguments
+ * @param {string} repo
+ * @param {string} owner
+ * @param {string[]} existingLabels
+ * @param {string} targetBranch
+ *
+ * @returns {Promise<{presentBlockingLabels: string[], missingRequiredLabels: string[]}>}
+ */
+export async function getPresentBlockingLabelsAndMissingRequiredLabels(
+  { github, context, core },
+  repo,
+  owner,
+  existingLabels,
+  targetBranch,
+) {
+  // this is a little bit of a strange looking branch format, but not going to touch this for now.
+  const repoTargetBranch = `${repo}/${targetBranch}`;
```

Comment:
todo: anywhere that I'm passing in this different format of `branch`, I'm going to rename the argument to `repoTargetBranch`.